support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/DefaultOAuth20RequestParameterResolver.java

```diff
@@ -272,4 +273,10 @@ public Set<String> resolveSupportedPromptValues(final String url) {
             .filter(supported::contains)
             .collect(Collectors.toSet()));
     }
+
+    @Override
+    public boolean isParameterOnQueryString(final WebContext context, final String name) {
+        val queryString = ((JEEContext) context).getNativeRequest().getQueryString();
+        return StringUtils.contains(queryString, name + '=');
```

Comment:
Will this cause problem for the following URL?

```
/oidc/something?hello=world&custom=https://google.com?client_secret=xyz
```

Reply:
I think you'll need to fetch the full URL first, including all the query parameters, and the use something like UriBuilder to only parse parameters on the URL that belong to CAS, and not everything that is part of the query string.

Reply:
Yes, you're right. There may be tricky URLs.

I think I have an even easier solution (I just committed) for that: https://github.com/apereo/cas/pull/5808/commits/47f29a1d902245897d4483af927d422d118a9849#diff-492188b377e5f1fa2f262414bdbd3743b15a2f876bb157e24bdfc316297905a5R280